python/pylibcudf/pylibcudf/libcudf/join.pxd

```diff
@@ -74,3 +77,113 @@ cdef extern from "cudf/join.hpp" namespace "cudf" nogil:
         const table_view left,
         const table_view right,
     ) except +
+
+    cdef gather_map_pair_type conditional_inner_join(
+        const table_view left,
+        const table_view right,
+        const expression binary_predicate,
+    ) except +
```

Comment:
Should we consider switching these apis to throw `cudf::data_type_error` at the c++ level and using `+libcudf_exception_handler` here? That way we can test for/throw `TypeError` on the pylibcudf side if the predicate doesn't return a boolean.

Reply:
Sure, done.